src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -1,15 +1,1401 @@
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_SWING_AXE);
+    } else if (this->skelAnime.animCurrentFrame == 21.0f) {
+        sp2C.x = this->actor.posRot.pos.x + Math_Sins(this->actor.shape.rot.y + 0x6A4) * 70.0f;
+        sp2C.z = this->actor.posRot.pos.z + Math_Coss(this->actor.shape.rot.y + 0x6A4) * 70.0f;
+        sp2C.y = this->actor.posRot.pos.y;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_HIT_GND);
+        func_8005AA1C(&globalCtx->mainCamera, 2, 0x19, 5);
+        func_800AA000(this->actor.xzDistFromLink, 0xFF, 0x14, 0x96);
+        func_80062CD4(globalCtx, &sp2C);
+    }
+
+    if ((17.0f < this->skelAnime.animCurrentFrame) && (this->skelAnime.animCurrentFrame < 23.0f)) {
+        this->unk_2FE = 1;
+    } else {
+        if ((this->unk_2FB != 0) && (this->skelAnime.animCurrentFrame < 10.0f)) {
+            Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 1, 0x5DC, 0);
+            this->actor.shape.rot.y = this->actor.posRot.rot.y;
+        }
+        this->unk_2FE = 0;
+    }
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_80A7506C(this);
+    }
+}
+
+void func_80A7506C(EnIk* this) {
+    f32 frames = SkelAnime_GetFrameCount(&D_060029FC.genericHeader);
+
+    this->unk_2FE = 0;
+    this->unk_2F9 = (s8)frames;
+    this->unk_2F8 = 7;
+    this->unk_2FF = this->unk_2FE;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060029FC, 1.0f, 0.0f, frames, 0, -4.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_PULLOUT);
+    EnIk_SetupAction(this, func_80A7510C);
+}
+
+void func_80A7510C(EnIk* this, GlobalContext* globalCtx) {
+    f32 frames;
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) || (--this->unk_2F9 == 0)) {
+        if (this->unk_2F8 == 8) {
+            func_80A7489C(this);
+        } else {
+            frames = SkelAnime_GetFrameCount(&D_06002538.genericHeader);
+            this->unk_2F8 = 8;
+            SkelAnime_ChangeAnim(&this->skelAnime, &D_06002538, 1.5f, 0.0f, frames, 3, -4.0f);
+        }
+    }
+}
+
+void func_80A751C8(EnIk* this) {
+    f32 frames = SkelAnime_GetFrameCount(&D_060033C4.genericHeader);
+
+    this->unk_2FF = 2;
+    this->unk_300 = 0;
+    this->unk_2F8 = 6;
+    this->actor.speedXZ = 0.0f;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060033C4, 0.0f, 0.0f, frames, 3, -6.0f);
+    this->unk_2FC = 0;
+    EnIk_SetupAction(this, func_80A75260);
+}
+
+void func_80A75260(EnIk* this, GlobalContext* globalCtx) {
+    f32 temp_f0;
+
+    this->unk_300 += 0x1C2;
+    temp_f0 = Math_Sins(this->unk_300);
+    this->skelAnime.animPlaybackSpeed = ABS(temp_f0);
+
+    if (this->skelAnime.animCurrentFrame > 11.0f) {
+        this->unk_2FF = 3;
+    }
+    if (((1.0f < this->skelAnime.animCurrentFrame) && (this->skelAnime.animCurrentFrame < 9.0f)) ||
+        ((13.0f < this->skelAnime.animCurrentFrame) && (this->skelAnime.animCurrentFrame < 18.0f))) {
```

Comment:
same here imo, but we dont really have a set convention on this for now so you dont have to